engine/src/lib/ui/window/window.cc

```diff
@@ -71,7 +71,7 @@ UIWIDGETS_API(char*) Window_defaultRouteName(Window* ptr) {
   const std::string routeName = ptr->client()->DefaultRouteName();
   size_t size = routeName.length() + 1;
   char* result = static_cast<char*>(malloc(size));
-  routeName.copy(result, size);
+  strcpy(result, routeName.c_str());
```

Comment:
please check the potential memory leak here: the pointer result seems to be impossible to be released any more.

I am not very sure though, according to this feed https://stackoverflow.com/questions/56181049/allocate-memory-in-a-c-dll-free-in-c-sharp, we may have to deallocate the memory by hand in c++

Reply:
In C# 【Window_freeDefaultRouteName(routeNamePtr); 】deallocates the memory by free() in c++.